### PR TITLE
89386 Update Page Titles (Telehealth Appointment Video and Phone - Part 3)

### PR DESCRIPTION
**Pull request by vbahinwillit**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**: Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
- _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
This PR updates the document title that displays in the browser tab.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#89386

## Testing done
Unit and e2e testing.

## Screenshots
Screenshot 2024-08-12 at 10 49 05 AM
Screenshot 2024-08-12 at 10 47 10 AM
Screenshot 2024-08-12 at 10 49 18 AM

---

Screenshot 2024-08-12 at 10 30 11 AM
Screenshot 2024-08-12 at 10 30 26 AM
Screenshot 2024-08-12 at 10 44 06 AM

---

Screenshot 2024-08-12 at 10 38 58 AM
Screenshot 2024-08-12 at 10 39 19 AM
Screenshot 2024-08-12 at 10 40 57 AM

---

Screenshot 2024-08-12 at 10 33 40 AM
Screenshot 2024-08-12 at 10 34 15 AM
Screenshot 2024-08-12 at 10 36 34 AM

## What areas of the site does it impact?
VAOS

### Video appointments (including GFE)

- [ ] Given a user is on a `Upcoming` `VA Video` details page
Then the browser tab title should read `Video appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Past` `VA Video` details page
Then the browser tab title should read `Past video appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Canceled` `VA Video` details page
Then the browser tab title should read `Canceled video appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

---

- [ ] Given a user is on a `Upcoming` `VA Video over GFE` details page
Then the browser tab title should read `Video appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Past` `VA Video over GFE` details page
Then the browser tab title should read `Past video appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Canceled` `VA Video over GFE` details page
Then the browser tab title should read `Canceled video appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

### Video at VA location appointments

- [ ] Given a user is on a `Upcoming` `Video at a VA location` details page
Then the browser tab title should read `Video appointment at VA location on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Past` `Video at a VA location` details page
Then the browser tab title should read `Past video appointment at VA location on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Canceled` `Video at a VA location` details page
Then the browser tab title should read `Canceled video appointment at VA location on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

### Video at an ATLAS location appointments

- [ ] Given a user is on a `Upcoming` `Video appointment at an ATLAS location` details page
Then the browser tab title should read `Video appointment at an ATLAS location on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Past` `Video appointment at an ATLAS location` details page
Then the browser tab title should read `Past video appointment at an ATLAS location on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Canceled` `Video appointment at an ATLAS location` details page
Then the browser tab title should read `Canceled video appointment at an ATLAS location on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

### Phone appointments

- [ ] Given a user is on a `upcoming` `phone appointment` details page
Then the browser tab title should read `Phone appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `past` `phone appointment` details page
Then the browser tab title should read `Past phone appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `canceled` `phone appointment` details page
Then the browser tab title should read `Canceled phone appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx [hunk 1]
```diff
@@ -69,28 +73,51 @@ export default function ConfirmedAppointmentDetailsPage() {
...
+            ? `${prefix} in-person appointment on`
+            : 'In-person appointment on';
+      }
+      if (isVideo) {
+        pageTitle = prefix
+          ? `${prefix} video appointment on`
+          : 'Video appointment on';
+        if (isClinicVideoAppointment(appointment)) {
+          pageTitle = prefix
+            ? `${prefix} video appointment at VA location on`
```
Comment (JunTaoLuo) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:102]: I believe the spec says `{prefix} video appointment at a VA location on` and this is missing the `a`.

**Review by JunTaoLuo — CHANGES_REQUESTED** [review]

**Comment by JunTaoLuo** [comment]
Hmm it's weird that the video appointment title updates didn't require any test updates so I'm assuming titles weren't being tested. In that case, can we add some unit tests for the page titles for the various video appointments like the ones we have for the in-person, phone, and cc appointments?

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:102]: Good catch! Thanks.

Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:102]: Actually, it states:

{prefix} video appointment at VA location on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs

Comment (JunTaoLuo) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:102]: Ah oops, I was reading the line above it:
image
But this is odd, I think it reads better with an `a` and the ATLAS appointments have `an` in both of the parts in the checklist. Not sure why there's a discrepancy for the "at VA location" appointments. Can we check with design on this?

Comment (outerpress) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:102]: Good catch, please do include the "a" - `Upcoming video at a VA location` - thanks!

Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx]: Done.

**Comment by JunTaoLuo** [comment]
@vbahinwillit it looks like some tests still aren't passing, but otherwise the changes look good! Can you let me know once you've fixed the tests and I'll approve it!

**Comment by vbahinwillit** [comment]
> @vbahinwillit it looks like some tests still aren't passing, but otherwise the changes look good! Can you let me know once you've fixed the tests and I'll approve it!

@JunTaoLuo - All tests passing now.

**Review by JunTaoLuo — APPROVED** [review]
Great work!

**Pull request merged**